lib/galaxy/webapps/galaxy/config_watchers.py

```diff
@@ -25,12 +33,22 @@ def __init__(self, app):
         self.data_manager_config_watcher = get_tool_conf_watcher(reload_callback=lambda: reload_data_managers(self.app))
         self.tool_data_watcher = get_tool_data_dir_watcher(self.app.tool_data_tables, config=self.app.config)
         self.tool_watcher = get_tool_watcher(self, app.config)
+        if self.app.is_job_handler:
+            self.job_rule_watcher = get_watcher(app.config, 'watch_job_rules', monitor_what_str='job rules')
+        else:
+            self.job_rule_watcher = get_watcher(app.config, '__invalid__')
         self.start()
 
     def start(self):
         [self.tool_config_watcher.watch_file(config) for config in self.tool_config_paths]
         [self.data_manager_config_watcher.watch_file(config) for config in self.data_manager_configs]
         [self.tool_data_watcher.watch_directory(tool_data_path) for tool_data_path in self.tool_data_paths]
+        for job_rules_directory in self.job_rules_paths:
+            self.job_rule_watcher.watch_directory(
+                job_rules_directory,
+                callback=partial(reload_job_rules, self.app),
+                recursive=True,
+                extensions=('.py',))
```

Comment:
Am I reading this correctly, does this only watch for changes in .py files? All of our rules are loaded through .yaml datasets.

Reply:
(And we'd rather not re-read the yaml file every time we check a job rule)

Reply:
So the rule would load and cache the YAML at module load time? Yeah, I can make this configurable, or, maybe just ignore `.py[oc]`?

Reply:
yeah that's how ours works now. That data structure is essentially static and tiny to hold in memory, no reason to go to disk before every job scheduling query.

Ignoring .py[oc] sounds good too!